zerver/lib/import_realm.py

```diff
@@ -1303,3 +1306,78 @@ def import_analytics_data(realm: Realm, import_dir: Path) -> None:
     re_map_foreign_keys(data, 'analytics_streamcount', 'stream', related_table="stream")
     update_model_ids(StreamCount, data, 'analytics_streamcount')
     bulk_import_model(data, StreamCount)
+
+    fix_datetime_fields(data, 'analytics_fillstate')
+    handle_fill_state(data, 'analytics_fillstate', realm)
+
+# To Handle FillState we have to compare the state of new (current) server to the old
+# (imported realm) server. If the new server is currently empty, we just import
+# FillState from the old server, if the old server is ahead, we delete any entries
+# from the imported realm's analytics tables and if the new server is ahead, then
+# we have to count stats just for the target Realm.
+def handle_fill_state(data: TableData, table: TableName, realm: Realm) -> None:
+    fill_state = FillState.objects.all()
+    if fill_state is None:
+        # Assuming server is empty.
+        # TODO: A better condition is required here.
+        update_model_ids(FillState, data, 'analytics_fillstate')
+        bulk_import_model(data, FillState)
+    else:
+        for items in data[table]:
+            COUNT_STATS = get_count_stats(realm)
+            stat = COUNT_STATS[items['property']]
+            current_fill_state = fill_state.filter(property = stat.property).first()
+
+            if stat.frequency == CountStat.HOUR:
+                time_increment = timedelta(hours=1)
+            elif stat.frequency == CountStat.DAY:
+                time_increment = timedelta(days=1)
+
+            items = check_imported_fill_state(items, stat, time_increment, realm)
+
+            if current_fill_state is None:
+                filters = {'property': stat.property, 'realm': realm}
```

Comment:
Let's add a comment explaining the reasoning in each of these cases, and have them return.  E.g. this case is for when the server we exported from is ahead of the server we're importing into; our fix is to just delete the newer entries that we just imported.